app/api/resources/metadata.py

```diff
@@ -14,39 +14,37 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
-
+import json
 import logging
 
-from aiohttp import ClientResponseError
 from fastapi import APIRouter, Request, responses
 from loguru import logger
+from aiohttp import ClientResponseError
 
 from api.error_response import response_error_handler
+from api.models.statistics import GenomeStatistics
+from core.config import GRPC_HOST, GRPC_PORT
 from core.logging import InterceptHandler
-import json
+
+from api.resources.grpc_client import GRPCClient
+from google.protobuf.json_format import MessageToDict
 
 logging.getLogger().handlers = [InterceptHandler()]
 
 router = APIRouter()
 
+logger.log("INFO", "Connecting to gRPC server on " + GRPC_HOST + ":" + str(GRPC_PORT))
+grpc_client = GRPCClient(GRPC_HOST, GRPC_PORT)
+
 
-@router.get("/statistics", name="statistics")
-async def get_metadata_statistics(request: Request):
-    """"""
-    sample_response = {
-        "assembly_name": "GRCh38.p13",
-        "common_name": "Human",
-        "genome_id": "a7335667-93e7-11ec-a39d-005056b38ce3",
-        "genome_tag": "grch38",
-        "image": "https://beta.ensembl.org/static/genome_images/homo_sapiens_38.svg",
-        "is_available": True,
-        "popular_order": 0,
-        "reference_genome_id": None,
-        "scientific_name": "Homo sapiens",
-    }
+@router.get(
+    "/genome/{genome_uuid}/stats", name="statistics")
+async def get_metadata_statistics(request: Request, genome_uuid: str):
     try:
-        return responses.Response(content=json.dumps(sample_response))
+        top_level_stats_dict = MessageToDict(grpc_client.get_statistics(genome_uuid))
 
+        genome_stats = GenomeStatistics(_raw_data=top_level_stats_dict["statistics"])
+        return responses.Response(json.dumps({"genome_stats": genome_stats.dict()}))
     except (ClientResponseError, Exception) as e:
-        logger.log("DEBUG", e)
-        return response_error_handler({"EROR": e})
+        logger.log("INFO", e)
+        return response_error_handler({"ERROR": e})
```

Comment:
Log says INFO were as the response_error_handler says ERROR. Do you think this should be consistent ?

Reply:
That's the response to web interface and the logger INFO prints the logs in output in different debug modes. For example the debug mode is False we will still see the INFO printed out in logs.

Reply:
@kamaldodiya I updated the API to give 500 internal server error which seems like an appropriate message as an API response and then we can investigate the logs for more detail.